lib/errorResponseInterceptor.ts

```diff
@@ -14,15 +14,26 @@
  * limitations under the License.
  */
 
+const { pipeline } = require('node:stream/promises');
+const { Writable, Readable } = require('node:stream');
+
 export function errorResponseInterceptor(axiosError) {
   if (
     axiosError.response && // must have a response
     axiosError.response.status >= 400 &&
     // must have data:
     axiosError.response.data &&
-    // must be a JSON
+    // must be a JSON request
     // which also implies Content-Type starts with application/json:
-    axiosError.config.responseType === 'json' &&
+    (axiosError.config.responseType === 'json' ||
+      // or a stream request with application/json
+      // that has had the response converted to an object
+      (axiosError.config.responseType === 'stream' &&
+        axiosError.response.headers['content-type'].startsWith(
+          'application/json'
+        ) &&
+        !(axiosError.response.data instanceof Readable) &&
+        !(typeof axiosError.response.data === 'string'))) &&
```

Comment:
similar to https://github.com/IBM/cloudant-node-sdk/pull/1663/files#r1830844281

Reply:
I think we're safe to assume that a string we see here will be a `string` primitive and not a `String` that is also an `Object` so removed in 17cb22ef.